src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -324,13 +339,27 @@ var ReactComponentTreeHook = {
     }
 
     var currentOwner = ReactCurrentOwner.current;
-    if (currentOwner && typeof currentOwner._debugID === 'number') {
-      var id = currentOwner && currentOwner._debugID;
-      info += ReactComponentTreeHook.getStackAddendumByID(id);
+    if (currentOwner) {
+      if (typeof currentOwner.tag === 'number') {
+        const fiber = ((currentOwner : any) : Fiber);
+        info += ReactComponentTreeHook.getStackAddendumByFiber(fiber);
+      } else if (typeof currentOwner._debugID === 'number') {
+        info += ReactComponentTreeHook.getStackAddendumByID(currentOwner._debugID);
+      }
     }
     return info;
   },
 
+  getStackAddendumByFiber(fiber : Fiber) : string {
+    var info = '';
+    var node = fiber;
+    do {
+      info += describeFiber(node);
+      node = node.return;
```

Comment:
It is possible for a single Fiber to have been rendered by two different parents. E.g. in your example the `props.children` may have bailed out when the parent switched. So at any given point it can be both.

I think it may just depend on context. In some context you may want the "current" Fiber path but in the case of the key warning you probably want the path you took to get to the update which will have the correct `.return`.